apps/reply/interface.html

```diff
@@ -0,0 +1,125 @@
+    <script>
+        const form = document.querySelector("#replyForm");
+        form.addEventListener("submit", (event) => {
+            event.preventDefault();
+            saveReply();
+        });
+        var replies = [];
+        var fetching = true; // Bit of a hack
+        var el = document.getElementById('items');
+        function onInit(device) {
+            fetching = true;
+            Util.readStorageJSON("replies.json", (arr) => {
+                if (arr) { 
+                    replies = replies.concat(arr);
+                    fetching = false;
+                    renderReplyList();
+                }
+            });
+        }
+
+        function toggleVisibility() {
+            let empty = document.getElementById("empty");
+            let loading = document.getElementById("loading");
+            if (replies.length == 0) {
+                empty.setAttribute("class", "d-block");
+                loading.setAttribute("class", "d-hide");
+            }
+            else if (fetching) {
+                loading.setAttribute("class", "d-block");
+                empty.setAttribute("class", "d-hide");
+            }
+            else {
+                loading.setAttribute("class", "d-hide");
+                empty.setAttribute("class", "d-hide");
+            }
+        }
+
+        function renderReplyList() {
+            toggleVisibility();
+            let table = document.getElementById("replyList");
+            table.innerHTML = "";
+            if (!fetching) {
+                for (var i = 0; i < replies.length; i++) {
+                    var reply = replies[i];
+                    var li = document.createElement("tr");
+                    var label = document.createElement("td");
+                    label.innerHTML = reply.text + "      ";
+                    var deleteButton = document.createElement("button");
+                    deleteButton.innerText = "Delete";
+                    deleteButton.setAttribute("onclick", `deleteReply(${i})`);
+                    deleteButton.setAttribute("class", "btn btn-error");
+                    li.appendChild(label);
+                    li.appendChild(deleteButton);
+                    table.appendChild(li);
+                }
+            }
+        }
+
+        function deleteReply(index) {
+            replies.splice(index, 1);
+            renderReplyList(replies);
+        }
+
+        function saveReply() {
+            var reply = {};
+            reply.text = document.getElementById('msg').value;
+            replies.push(reply);
+            return updateDevice();
+        }
+
+        function updateDevice() {
+            fetching = true;
+            renderReplyList();
+            Util.writeStorage("replies.json", JSON.stringify(replies), () => {
+                fetching = false;
+                renderReplyList();
+                return true;
+            });
+            fetching = false;

```

Comment:
Should the `fetching = false` above (line 115) be the one to set `fetching` to false, and not this line?

Reply:
Yes you're right, fetching should only update when the promise resolves - fixed!